app/models/user.rb

```diff
@@ -0,0 +1,20 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable,
+         :omniauthable, omniauth_providers: [:google_oauth2]
+
+  def self.from_omniauth(access_token)
```

Comment:
We could clean this up a bit:

```ruby
def self.from_omniauth(access_token)
  data = access_token.info
  user = User.find_by(email: data['email'])
  return user if user.present?

  User.create(email: data['email'], password: Devise.friendly_token[0,20])
end
```

Reply:
oops. just cleaned it up, but the have seen your comment:)
agree, thank you!